src/main/java/net/coderbot/patchwork/at/AccessorInterfaceGenerator.java

```diff
@@ -0,0 +1,90 @@
+package net.coderbot.patchwork.at;
+
+import org.objectweb.asm.*;
+
+public class AccessorInterfaceGenerator {
+
+	public static void generate(ModGutter.Meta meta, ClassWriter writer) {
+		// Magic to get the parent class
+
+		writer.visit(Opcodes.V1_8,
+				Opcodes.ACC_PUBLIC | Opcodes.ACC_ABSTRACT | Opcodes.ACC_INTERFACE,
+				"patchwork_generated/mixin/" + meta.getName() + "Accessor",
+				null,
+				"java/lang/Object",
+				null);
+		//@Mixin(targetclass.class)
+		AnnotationVisitor mixinAnnotationVisitor =
+				writer.visitAnnotation("Lorg/spongepowered/asm/mixin/Mixin;", false);
+		{
+			AnnotationVisitor mixinAnnotationValueVisitor =
+					mixinAnnotationVisitor.visitArray("value");
+			mixinAnnotationValueVisitor.visit(null,
+					Type.getType("L" + meta.getAccessTransformerEntry().getClazzName() + ";"));
+			mixinAnnotationValueVisitor.visitEnd();
+		}
+		mixinAnnotationVisitor.visitEnd();
+		if(!meta.getDescriptor().contains("(") /* it's probably not a field*/) {
+			if((meta.getOpcode() & Opcodes.ACC_STATIC) != 0) /*if is static*/ {
+				MethodVisitor getter = writer.visitMethod(Opcodes.ACC_PUBLIC | Opcodes.ACC_STATIC,
+						"get" + meta.getName(),
+						"()" + meta.getName() + ";",
+						null,
+						null);
+				// Mark getter with @Accessor
+				getter.visitAnnotation("Lorg/spongepowered/asm/mixin/gen/Accessor;", true)
+						.visitEnd();
+				// Generate boilerplate that will be replaced by mixin stuff later todo might be
+				// unnecessary
+				getter.visitCode();
+				getter.visitLabel(new Label());
+				getter.visitInsn(Opcodes.ACONST_NULL);
```

Comment:
The class verifier might complain that this isn't well formed (in case of non object-reference return types). Consider just throwing an exception